config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/JmsDestinationEntityBuilder.java

```diff
@@ -222,15 +224,21 @@ private void buildInboundDestination (
         ServiceResolutionSettings serviceResolutionSettings = inboundDetail.getServiceResolutionSettings();
         if (serviceResolutionSettings != null) {
             String serviceRef = serviceResolutionSettings.getServiceRef();
-            if (serviceRef != null) {
-                Service service = bundle.getServices().get(serviceRef);
-                if (service == null) {
+            Service service = bundle.getServices().get(serviceRef);
+
+            if (service == null) {
+                if (getDeploymentBundle() == null) {
                     throw new EntityBuilderException("Could not find associated Service for inbound JMS Destination: " + name + ". Service Path: " + serviceRef);
+                } else {
+                    service = getDeploymentBundle().getServices().get(serviceRef.substring(serviceRef.lastIndexOf('/') + 1));
                 }
+            }
+
+            if (service == null) {
```

Comment:
If `serviceRef` is not null, but the corresponding `service` is not found in the bundle, then this is an error.